internal/fnruntime/container.go

```diff
@@ -158,20 +161,29 @@ func NewContainerEnvFromStringSlice(envStr []string) *runtimeutil.ContainerEnv {
 // exist.
 func (f *ContainerFn) prepareImage() error {
 	// check image existence
-	args := []string{"image", "ls", f.Image}
+	foundImageInLocalCache := false
+	args := []string{"image", "inspect", f.Image}
 	cmd := exec.Command(dockerBin, args...)
 	var output []byte
 	var err error
-	if output, err = cmd.CombinedOutput(); err != nil {
-		return &ContainerImageError{
-			Image:  f.Image,
-			Output: string(output),
-		}
-	}
-	if strings.Contains(string(output), strings.Split(f.Image, ":")[0]) {
+	if _, err = cmd.CombinedOutput(); err == nil {
 		// image exists locally
+		foundImageInLocalCache = true
+	}
+
+	// If env var ImagePullPolicyEnv is set to "ifnotpresent", we scan the local
+	// images first. If there is a match, we just return. This can be useful for
+	// local development to prevent the remote image to accidentally override
+	// the local image when they use the same name and tag.
+	if strings.ToLower(os.Getenv(ImagePullPolicyEnv)) == "ifnotpresent" && foundImageInLocalCache {
```

Comment:
Why use an env var instead of a flag which provides better documentation? This is also consistent with how other container-related knobs are exposed

Reply:
This behavior will affect both `render` and `eval`. And currently `eval` exposes these container-related knobs, but `render` doesn't. We of course can add a new flag for `eval`. But I want to ensure adding a new flag makes sense for `render` before doing that.
@frankfarzan @droot @Shell32-Natsu Thoughts?

Reply:
Yes, add a flag to both.